operator/pkg/central/extensions/reconcile_admin_password.go

```diff
@@ -113,9 +115,11 @@ func (r *reconcileAdminPasswordExtensionRun) updateStatus(status *platform.Centr
 	if status.Central.AdminPassword == nil {
 		status.Central.AdminPassword = &platform.AdminPasswordStatus{}
 	}
-	if r.infoUpdate == status.Central.AdminPassword.Info {
+	if r.infoUpdate == status.Central.AdminPassword.Info && r.passwordSecretName == *status.Central.AdminPassword.SecretReference {
 		return false
 	}
+
+	status.Central.AdminPassword.SecretReference = &r.passwordSecretName
```

Comment:
This worries me a little. I'm somewhat afraid that (given how this whole codebase, and **especially** status updates is made of closures upon closures) some refactoring elsewhere might cause `r.passwordSecretName` to get changed beneath our feet **before** `status` is serialized. 

How about:

```suggestion
        secretNameCopy := r.passwordSecretName
	status.Central.AdminPassword.SecretReference = &secretNameCopy
```